# Homework 2

### Importing the libraries into python

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

### Importing the dataset into python

In [4]:
df = pd.read_csv("hw2_dataset.csv")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 969 entries, 0 to 968
Columns: 1025 entries, ID to D_1023
dtypes: int64(1024), object(1)
memory usage: 7.6+ MB


### Preprocessing the dataset
From the df.info() function I am able to find out that most of the data inside the document are in the form of integers with one of them being a charater or string column

In [6]:
df.head()

ID  D_0  D_1  D_2  D_3  D_4  D_5  D_6  D_7  D_8  \
0  AAEAMMIUQZAASJ-MRXNPFEDSA-N    1    1    1    1    1    1    1    0    1   
1  AAEFNWQXBPYXAC-UHFFFAOYSA-N    1    1    1    1    0    1    1    1    0   
2  AAMHSIWFDKXUMZ-UHFFFAOYSA-N    1    1    1    1    1    1    0    1    1   
3  AAPQXEOSVSLLMB-UHFFFAOYSA-N    1    1    1    1    0    1    1    1    1   
4  AARXXEHXOBTROW-UHFFFAOYSA-N    1    1    1    1    1    1    1    0    1   

   ...  D_1014  D_1015  D_1016  D_1017  D_1018  D_1019  D_1020  D_1021  \
0  ...       1       1       1       1       1       1       0       1   
1  ...       1       0       1       1       1       0       0       1   
2  ...       1       1       1       1       1       1       0       0   
3  ...       1       1       1       1       1       1       1       1   
4  ...       1       1       1       1       1       1       0       1   

   D_1022  D_1023  
0       1       1  
1       1       1  
2       1       1  
3       1       1  
4       1       1  

[5 rows x 1025 columns]

From performing df.head it can be seen that the column that has characters is the ID column. I will now find the number of unique elements inside that column to check whether all the items in that column are unique if they are unique then that means that they will not be a great addition to the prediction model.

In [7]:
df.ID.nunique()

969

As there are 969 unique variables in the ID column it can be said that we can actually reduce that column to make it a cleaner dataset